### ISSUE TITLE
Internal Server Error when performing authorization_code grant

### ISSUE THREAD

**Issue by anoopgarlapati** [spring-projects-experimental/spring-authorization-server#222]
**Describe the bug**
When performing authorization_code flow, the authorize endpoint throws 500 Internal Server Error.

**To Reproduce**
Run sample application as documented in samples README. Attempting to login the user fails with 500 error.

**Expected behavior**
User login should succeed and an authorization_code should be received.

**Sample**
The sample included in the repository currently fails.

The exception stacktrace from authorization-server logs:
```
2021-02-10 14:54:35.278 ERROR 93708 --- [nio-9000-exec-9] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception

java.lang.IllegalArgumentException: The authorization must be unique. Found duplicate identifier: b622a210-6063-467f-8cc6-5ae91b467978
        at org.springframework.util.Assert.isTrue(Assert.java:121) ~[spring-core-5.3.1.jar:5.3.1]
        at org.springframework.security.oauth2.server.authorization.InMemoryOAuth2AuthorizationService.save(InMemoryOAuth2AuthorizationService.java:46) ~[spring-security-oauth2-authorization-server-0.1.0-SNAPSHOT.jar:0.1.0-SNAPSHOT]
        at org.springframework.security.oauth2.server.authorization.web.OAuth2AuthorizationEndpointFilter.processUserConsent(OAuth2AuthorizationEndpointFilter.java:274) ~[spring-security-oauth2-authorization-server-0.1.0-SNAPSHOT.jar:0.1.0-SNAPSHOT]
        at org.springframework.security.oauth2.server.authorization.web.OAuth2AuthorizationEndpointFilter.doFilterInternal(OAuth2AuthorizationEndpointFilter.java:155) ~[spring-security-oauth2-authorization-server-0.1.0-SNAPSHOT.jar:0.1.0-SNAPSHOT]
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.3.1.jar:5.3.1]
```

**Comment by Kehrlann** [comment]
Just ran into this issue as well.

I think it'd be a good thing to have an end-to-end test through the API for the authorization code flow ; that does and `/authorize` call, capture the issued `code`, and then do a `/token` call with that code.

Thoughts on this ?

**Comment by jgrandja** [comment]
Thanks @anoopgarlapati @Kehrlann ! Obviously test coverage was missing here :(

I will add a fix and test within the next couple of hours.

**Comment by nickmelis** [comment]
Having the same issue here @jgrandja

**Comment by jgrandja** [comment]
@anoopgarlapati @Kehrlann @nickmelis The implementation of `InMemoryOAuth2AuthorizationService.save()` was simply wrong. It has been fixed and merged via 09846ee

**Comment by nickmelis** [comment]
@jgrandja just gave it an update, and it works beautifully 🥇
